@here/harp-mapview/lib/copyrights/CopyrightProvider.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoBox } from "@here/harp-geoutils";
+import { getOptionValue, LoggerManager } from "@here/harp-utils";
+import { CopyrightInfo } from "./CopyrightInfo";
+
+declare const require: any;
+
+// tslint:disable-next-line:no-var-requires
+const RBush = require("rbush");
+
+/**
+ * `CopyrightProvider` is an abstract class to retrieve copyrights information for geographic region
+ * specified by bounding box.
+ */
+export abstract class CopyrightProvider {
```

Comment:
Please separate `CopyrightProvider` interface (and make it interface) from CopyrightCoverageProvider and Url..Provider. I mean to separate files.
Random default implementation shouldn't be required to even import an interface.

Reply:
Done.